data/tutorials/guides/rs_02_comparison_of_standard_containers.md

```diff
@@ -0,0 +1,102 @@
+---
+id: data-structures-comparison 
+title: Comparison of Standard Containers
+description: >
+  Rough comparison of the different container types in OCaml
+category: "Resources"
+---
+
+# Comparison of Standard Containers
+This is a rough comparison of the different container types that are
+provided by the OCaml language or by the OCaml standard library. In each
+case, n is the number of valid elements in the container.
+
+Note that the big-O cost given for some operations reflects the current
+implementation but is not guaranteed by the official documentation.
+Hopefully it will not become worse. Anyway, if you want more details,
+you should read the documentation about each of the modules. Often, it
+is also instructive to read the corresponding implementation.
+
+## Lists: immutable singly-linked lists
+Adding an element always creates a new list l from an element x and list
+tl. tl remains unchanged, but it is not copied either.
+
+* "adding" an element: O(1), cons operator `::`
+* length: O(n), function `List.length`
+* accessing cell `i`: O(i)
+* finding an element: O(n)
+
+Well-suited for: IO, pattern-matching
+
+Not very efficient for: random access, indexed elements
+
+## Arrays: mutable vectors
+Arrays are mutable data structures with a fixed length and random access.
+
+* "adding" an element (by creating a new array): O(n)
+* length: O(1), function `Array.length`
+* accessing cell `i`: O(1)
+* finding an element: O(n)
+
+Well-suited for sets of elements of known size, access by numeric index,
+in-place modification. Basic arrays have a fixed length.
```

Comment:
I'm a little confused by these sentences. Do you mean that Arrays are well-suited...., can access (or be accessed>)..., have in-place modification?